app/public/js/localSprites.js

```diff
@@ -217,17 +253,88 @@ function addFireballContrailParticle(parent, decayFrames, parentPreScalingXSize,
     } else {
         randomY = Math.round(parent.y - ((Math.random() * parent.yScale * parentPreScalingYSize) / 2));
     }
-    var fireballContrailParticle = new SimpleSprite({frames}, randomX, randomY, 0, 0, 1.25, 2.5);
-    fireballContrailParticle.framesToLive = decayFrames;
-    fireballContrailParticle.scaleOscillation = true;
-    fireballContrailParticle.xScalePerFrame = fireballContrailParticle.xScale / fireballContrailParticle.framesToLive;
-    fireballContrailParticle.yScalePerFrame = fireballContrailParticle.yScale / fireballContrailParticle.framesToLive;
-    fireballContrailParticle.xScaleMin = 0;
-    fireballContrailParticle.yScaleMin = 0;
-    fireballContrailParticle.rotationPerFrame = 50;
-    //fireballContrailParticle.msBetweenFrames = Math.round((decayFrames * 50 /*or framerate*/) / frames.length) + 1; //'+1' hopefully keeps the animation from starting to loop just before the pariticle dies.  //would be better to also have a continuous alpha fade happen during this time. Could also scale down if that weren't build into the animation frames already.
-    //parent.contrailParticles.push(fireballContrailParticle);
-    localSprites.push(fireballContrailParticle); //BROKEN: Should push to parent.contrailParticles, but then render.js should render things in that array. I don't know the syntax for that yet, I don't think.
+    var fireballParticle = new SimpleSprite({frames}, randomX, randomY, 0, 0, 1.25, 2.5);
+    if (type === 0) fireballParticle.name = 'fireballContrailParticle';
```

Comment:
Let's make these consts too, and here you can remove the condition and just do:
`fireballParticle.type = type`
In fact, once we have this, we could just make this function generic like `addChildSprite` and it could be used for things besides fireballs and fireball particles, since you are passing in what type of child to create.